packages/stream-model-instance-handler/src/__tests__/model-instance-document-handler.test.ts

```diff
@@ -739,6 +803,112 @@ describe('ModelInstanceDocumentHandler', () => {
+    )) as SignedCommitContainer
+    await ipfs.dag.put(signedCommitFail, FAKE_CID_2)
+
+    const payloadFail = dagCBOR.decode(signedCommitFail.linkedBlock)
+    await ipfs.dag.put(payloadFail, signedCommitFail.jws.link)
+
+    const signedCommitDataFail = {
+      cid: FAKE_CID_2,
+      type: CommitType.SIGNED,
+      commit: payloadFail,
+      envelope: signedCommitFail.jws,
+    }
+    await expect(handler.applyCommit(signedCommitDataFail, context, state)).rejects.toThrow(
+      'Unique content fields value does not match metadata. If you are trying to change the value of these fields, this is causing this error: these fields values are not mutable.'
+    )
+
+    const signedCommitOK = (await ModelInstanceDocument.makeUpdateCommit(
+      context.signer,
+      doc.commitId,
+      doc.content,
+      { one: 'foo', two: 'bar', myData: 2 }
+    )) as SignedCommitContainer
+    await ipfs.dag.put(signedCommitOK, FAKE_CID_3)
+
+    const payloadOK = dagCBOR.decode(signedCommitOK.linkedBlock)
+    await ipfs.dag.put(payloadOK, signedCommitOK.jws.link)
+
+    const signedCommitDataOK = {
+      cid: FAKE_CID_3,
+      type: CommitType.SIGNED,
+      commit: payloadOK,
+      envelope: signedCommitOK.jws,
+    }
+    state = await handler.applyCommit(signedCommitDataOK, context, state)
+    expect(state).toMatchSnapshot()
+  })
+
+  it('MIDs with SET account relation validate content schema on update', async () => {
+    const genesisCommit = (await ModelInstanceDocument._makeGenesis(
+      context.signer,
+      null,
+      { ...DETERMINISTIC_METADATA, model: FAKE_MODEL_SET_ID },
+      ['a', 'b']
+    )) as GenesisCommit
+    await ipfs.dag.put(genesisCommit, FAKE_CID_1)
+
+    // apply genesis
+    const genesisCommitData = {
+      cid: FAKE_CID_1,
+      type: CommitType.GENESIS,
+      commit: genesisCommit,
+    }
+
+    // The deterministic genesis creation works independently of content validation as determinitic commits have no content
+    const state = await handler.applyCommit(genesisCommitData, context)
+    const state$ = TestUtils.runningState(state)
+    const doc = new ModelInstanceDocument(state$, context)
+
+    const signedCommit = (await ModelInstanceDocument.makeUpdateCommit(
+      context.signer,
+      doc.commitId,
+      doc.content,
+      { one: 'a', two: 'b', myData: 2 }
+    )) as SignedCommitContainer
+    await ipfs.dag.put(signedCommit, FAKE_CID_2)
+
+    const payload = dagCBOR.decode(signedCommit.linkedBlock)
+    await ipfs.dag.put(payload, signedCommit.jws.link)
+
+    const signedCommitData = {
+      cid: FAKE_CID_2,
+      type: CommitType.SIGNED,
+      commit: payload,
+      envelope: signedCommit.jws,
+    }
+    // MID can't be updated with content that does not pass schema validation
+    await expect(handler.applyCommit(signedCommitData, context, state)).rejects.toThrow(
+      'Validation Error: data/one must NOT have fewer than 2 characters, data/two must NOT have fewer than 2 characters'
```

Comment:
what is data/one and data/two ?  I think this would be a confusing error message to me as a developer

Reply:
This is the error message thrown by AJV (the JSON schema validation library), we can customize them I think but that would likely need a dedicated project.